Test/AblyTests/Tests/MessageUpdatesDeletesTests.swift

```diff
@@ -0,0 +1,396 @@
+import Ably
+import Nimble
+import XCTest
+
+@testable import Ably
+
+class MessageUpdatesDeletesTests: XCTestCase {
+    
+    // MARK: - Test Environment
+    
+    private enum TestEnvironment {
+        case rest(client: ARTRest, testHTTPExecutor: TestProxyHTTPExecutor, channelName: String)
+        case realtime(client: ARTRealtime, testHTTPExecutor: TestProxyHTTPExecutor, channelName: String)
+        
+        var channel: ARTChannelProtocol {
+            switch self {
+            case .rest(let client, _, let channelName):
+                return client.channels.get(channelName)
+            case .realtime(let client, _, let channelName):
+                return client.channels.get(channelName)
+            }
+        }
+        
+        var requests: [URLRequest] {
+            switch self {
+            case .rest(_, let testHTTPExecutor, _),
+                 .realtime(_, let testHTTPExecutor, _):
+                return testHTTPExecutor.requests
+            }
+        }
+        
+        static func rest(_ test: Test, channelPrefix: String = "mutable:") throws -> TestEnvironment {
```

Comment:
why do we need a channel prefix?

Reply:
It's form JS, without it editing of messages doesn't work. I've removed it as a parameter though since this class is probably not going to end up being very reusable.